### PR TITLE
Added a DashboardAggregator

### PR DESCRIPTION
**Pull request by Graeme-Miller**
Added a DashboardAggregator. This aggregator will pull together dashboard sensors from it's children and merge them together. This however, does not touch the owner or environment sensors as there is no good way to aggregate them.

### CODE REVIEW

**Comment by aledsage** [comment]
Test failure looks real:
```
java.lang.IllegalStateException: Management context no longer running
	at org.apache.brooklyn.enricher.stock.aggregator.AggregationJobTest.testSetup(AggregationJobTest.java:55)
```

**Comment by Graeme-Miller** [comment]
test failure fixed, only thing left to decide is whether this is the correct place or not

File: core/src/main/java/org/apache/brooklyn/enricher/stock/aggregator/AggregationJob.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+        //Locations merge
+        //We are merging a Map, of Lists of Maps
+        //The outer map is location type, e.g. servers to a list of data for that type
+        //The inner map should contain the data e.g. name/icon/count.
+
+        //Further complicating this is that the Map you get back from sensors/config doesn't conform to the generic expectations.
+        //I.E. even if you type this as Map<String, List<Map<String, String>>> you will still get back an integer from
+        //the inner map- hence all the weird casting bellow.
+        Map<String, List<Map<String, Object>>> outerLocationMapFromEntity = entity.sensors().get(DASHBOARD_LOCATIONS);
```
[the remark this answers is not in the document]
Comment (Graeme-Miller) [reply core/src/main/java/org/apache/brooklyn/enricher/stock/aggregator/AggregationJob.java]: That hasn't been agreed yet, but your suggestion seems sensible

**Comment by aledsage** [comment]
Thanks @Graeme-Miller - merging now.

**Pull request merged**